Modules/Devices/src/Devices.ConsumerApi/Controllers/DevicesController.cs

```diff
@@ -77,18 +77,17 @@ public async Task<IActionResult> GetActiveDevice()
 
     [HttpDelete("{id}")]
     [ProducesResponseType(StatusCodes.Status200OK)]
-    [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
     public async Task<IActionResult> DeleteDevice([FromRoute] DeviceId id, CancellationToken cancellationToken)
     {
-        await _mediator.Send(new DeleteDeviceCommand { DeviceId = id, }, cancellationToken);
-
-        return NoContent();
+        try
+        {
+            await _mediator.Send(new DeleteDeviceCommand { DeviceId = id, }, cancellationToken);
+            return NoContent();
+        }
+        catch (NotFoundException ex)
```

Comment:
It's not necessary to catch the exception. This is done by the exception filter (Backbone.BuildingBlocks.API.Mvc.ExceptionFilters.CustomExceptionFilter), which automatically converts a NotFoundException to a 404. Similar to ApplicationException or DomainException.